### PR TITLE
Reverse a linked list in python program

### PR DESCRIPTION
**Pull request by poojarathore30**
### **Describe your change:**
* [✔ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [✔ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [✔ ] This pull request is all my own work -- I have not plagiarized.
* [✔ ] I know that pull requests will not be merged if they fail the automated tests.
* [✔ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [✔ ] All new Python files are placed inside an existing directory.
* [✔ ] All filenames are in all lowercase characters with no spaces or dashes.
* [✔ ] All functions and variable names follow Python naming conventions.
* [✔ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [✔ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ✔] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [✔ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: data_structures/linked_list/reverse_linked_list_program.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Time Complexity : O(n) 
+# Space Complexity : O(1) 
+
```
Comment (dhruvmanila) [inline data_structures/linked_list/reverse_linked_list_program.py:3]: If possible, please provide a `docstring` explaining what this algorithm is about and provide some additional resources where people can go to learn more.

File: data_structures/linked_list/reverse_linked_list_program.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		self.next = None
+
+class LinkedList: 
+
+	# Function to initialize head 
+	def __init__(self): 
+		self.head = None
+
+	# Function to reverse the linked list 
+	def reverse(self):
```
Comment (dhruvmanila) [inline data_structures/linked_list/reverse_linked_list_program.py:19]: Type hints, `docstring` and `doctest` for the function

File: data_structures/linked_list/reverse_linked_list_program.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+# Time Complexity : O(n) 
+# Space Complexity : O(1) 
+
+# Node class 
+class Node: 
+
+	# Constructor to initialize the node object 
+	def __init__(self, data):
```
Comment (dhruvmanila) [inline data_structures/linked_list/reverse_linked_list_program.py:8]: type hints

File: data_structures/linked_list/reverse_linked_list_program.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+	# Function to initialize head 
+	def __init__(self): 
+		self.head = None
+
+	# Function to reverse the linked list 
+	def reverse(self): 
+		prev = None
+		current = self.head 
+		while(current is not None):
```
Comment (dhruvmanila) [inline data_structures/linked_list/reverse_linked_list_program.py:22]:
```suggestion
		while current: 
```
If `current` variable is going to become `None` in the future, `while` loop will exit.

File: data_structures/linked_list/reverse_linked_list_program.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+		current = self.head 
+		while(current is not None): 
+			next = current.next
+			current.next = prev 
+			prev = current 
+			current = next
+		self.head = prev 
+		
+	# Function to insert a new node at the beginning 
+	def push(self, new_data):
```
Comment (dhruvmanila) [inline data_structures/linked_list/reverse_linked_list_program.py:30]: Type hints, `docstring`, and `doctest` for the function

File: data_structures/linked_list/reverse_linked_list_program.py [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+		self.head = prev 
+		
+	# Function to insert a new node at the beginning 
+	def push(self, new_data): 
+		new_node = Node(new_data) 
+		new_node.next = self.head 
+		self.head = new_node 
+
+	# Utility function to print the linked LinkedList 
+	def printList(self):
```
Comment (dhruvmanila) [inline data_structures/linked_list/reverse_linked_list_program.py:36]:
```suggestion 
	def print_list(self): 
	"""Utility function to print the linked LinkedList"""
```
The function name should be in `snake_case` as discussed in CONTRIBUTING.md

Please add `doctest` and type hints to this function

File: data_structures/linked_list/reverse_linked_list_program.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+llist.push(4) 
+llist.push(15) 
+llist.push(85) 
+
+print("Given Linked List")
+llist.printList() 
+llist.reverse() 
+print("\nReversed Linked List")
+llist.printList() 
+
```
Comment (dhruvmanila) [inline data_structures/linked_list/reverse_linked_list_program.py:55]: This should be under `if __name__ == "__main__":` block

Also, your entire code is in 8 spaces while Python uses only 4 spaces as indentation.

**Comment by poyea** [comment]
I think there's an existing implementation on this: https://github.com/TheAlgorithms/Python/blob/78eca843e238dd65504f5f7fc2774e674243b215/data_structures/linked_list/singly_linked_list.py#L146. You may improve upon that or similar algorithms in the folder.

**Pull request closed without merging**